src/Core/functions.php

```diff
@@ -254,10 +254,28 @@ function env($key, $default = null)
      * Helper method for outputting deprecation warnings
      *
      * @param string $message The message to output as a deprecation warning.
+     * @param int $stackFrame The stack frame to include in the error. Defaults to 2
+     *   as that should point to application/plugin code.
      * @return void
      */
-    function deprecationWarning($message)
+    function deprecationWarning($message, $stackFrame = 2)
     {
+        if (!Configure::read('debug')) {
```

Comment:
If we do this, then we wont be able to find any left overs in production.

Reply:
Yes, but I'm not comfortable adding potentially 100s of calls to `debug_backtrace()` to people's applications in production mode. Would it be better to check the error level here instead?

Reply:
Yeah, I think so.

Reply:
I think we should have a separate flag for this. Something along `log3xDeprecations` (default true if not set), which should go into a dedicated log file. That way it is easy to spot in production and easy to disable if you just want to run your app and not update it.